src/main/java/org/sagebionetworks/EvaluationUtils.java

```diff
@@ -142,6 +168,84 @@ public static boolean isBooleanAnnotationPresentAndTrue(SubmissionStatus status,
 		return false;
 	}
 
+	public static void setAnnotationStringV2(SubmissionStatus status, String key, String value, boolean isPrivate) {
+		org.sagebionetworks.repo.model.annotation.v2.Annotations annotations = status.getSubmissionAnnotations();
+		if (annotations == null) {
+			annotations = new org.sagebionetworks.repo.model.annotation.v2.Annotations();
+			status.setSubmissionAnnotations(annotations);
+		}
+		if (annotations.getAnnotations() == null) {
+			annotations.setAnnotations(new HashMap<String, AnnotationsValue>());
+		}
+
+		if (annotations.getAnnotations().containsKey(key)) {
+			AnnotationsValue annotationsValue  = annotations.getAnnotations().get(key);
+			annotationsValue.getValue().clear();
+			annotationsValue.setType(AnnotationsValueType.STRING);
+			annotationsValue.getValue().add(value);
+			annotations.getAnnotations().put(key, annotationsValue);
+		} else {
+			AnnotationsValue annotationsValue = new AnnotationsValue();
+			annotationsValue.setType(AnnotationsValueType.STRING);
+			List<String> stringList = new ArrayList<>();
+			stringList.add(value);
+			annotationsValue.setValue(stringList);
+			annotations.getAnnotations().put(key, annotationsValue);
+		}
+	}
+
+	public static void setAnnotationLongV2(SubmissionStatus status, String key, Long value, boolean isPrivate) {
```

Comment:
The only difference between this methods is the type of the annotation and the value type in input, you can create a single method that handles this generically and takes in input a string as value plus the annotation type as the logic will be the same.

Reply:
E.g. setAnnotationV2(status, key, AnnotationsValueType.LONG, value.toString());